### PR TITLE
[CP-2039] I want to be able to read and accept the Privacy policy the first time the application is launched after it has been updated

### PR DESCRIPTION
**Pull request by lkowalczyk87**
Jira: [CP-2039]

**Description**

<details>
<summary><b>Screenshots</b></summary>

image

</details>

[CP-2039]: https://appnroll.atlassian.net/browse/CP-2039?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: out.txt [hunk 1]
```diff
@@ -0,0 +1,139 @@
+
```
Comment (OskarMichalkiewicz) [inline out.txt:1]: I guess this should be deleted 😁

File: packages/app/src/modals-manager/components/modals-manager.component.tsx [hunk 2]
```diff
@@ -28,6 +31,14 @@ const ModalsManager: FunctionComponent<Props> = ({
...
+  const { privacyPolicyAccepted } = useSelector(
+    (state: ReduxRootState) => state.settings
+  )
+
+  if (privacyPolicyAccepted === false) {
+    return <PrivacyPolicyModal />
+  }
+
   return (
     <>
```
Comment (OskarMichalkiewicz) [inline packages/app/src/modals-manager/components/modals-manager.component.tsx:43]:
```suggestion
    <>
        {!privacyPolicyAccepted && <PrivacyPolicyModal />}
```

File: out.txt (hunk 1 above)
Comment (lkowalczyk87) [reply out.txt]: deleted ;)

**Review by OskarMichalkiewicz — DISMISSED** [review]

**Review by OskarMichalkiewicz — APPROVED** [review]

File: packages/app/src/settings/components/privacy-policy-modal/privacy-policy-modal.component.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  }, [collectingData, dispatch])
+
+  if (collectingData === undefined) {
+    return null
+  }
+
+  return (
+    <ModalDialog
+      open
+      zIndex={200}
```
Comment (patryk-sierzega) [inline packages/app/src/settings/components/privacy-policy-modal/privacy-policy-modal.component.tsx:84]: I know this is fine, but in the future we should avoid doing, cause every modal has it's own setting of zIndex. It should be managed by modal manager imho :)

File: packages/app/src/settings/services/settings.service.ts [hunk 4]
```diff
@@ -32,7 +32,11 @@ export class SettingsService {
   }
 
   updateSettings({ key, value }: SettingsUpdateOption): SettingsValue {
-    this.store.set(key, value)
+    if (value !== undefined) {
+      this.store.set(key, value)
+    } else {
+      this.store.delete(key)
+    }
```
Comment (patryk-sierzega) [inline packages/app/src/settings/services/settings.service.ts:39]: Maybe update should only update and another function/method should be delivered to do deleting?

File: packages/app/src/settings/actions/delete-collecting-data.action.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import { createAsyncThunk } from "@reduxjs/toolkit"
+import { SettingsEvent } from "App/settings/constants"
+import { updateSettings } from "App/settings/requests"
+
+export const deleteCollectingData = createAsyncThunk(
+  SettingsEvent.DeleteCollectingData,
+  async () => {
+    await updateSettings({ key: "collectingData", value: undefined })
+
+    return
```
Comment (patryk-sierzega) [inline packages/app/src/settings/actions/delete-collecting-data.action.ts:15]: Do you need this return without value? I think js/ts will return undefined when there is no return :)

File: packages/app/src/modals-manager/components/modals-manager.component.tsx [hunk 6]
```diff
@@ -28,6 +31,14 @@ const ModalsManager: FunctionComponent<Props> = ({
   deviceInitializationFailedModalShowEnabled,
   hideModals,
 }) => {
+  const { privacyPolicyAccepted } = useSelector(
+    (state: ReduxRootState) => state.settings
+  )
+
+  if (privacyPolicyAccepted === false) {
+    return <PrivacyPolicyModal />
+  }
```
Comment (patryk-sierzega) [inline packages/app/src/modals-manager/components/modals-manager.component.tsx:40]: a bit nit-picking ;D i've got mixed feelings about value === false or value === true, maybe '!privacyPolicyAccepted' would be better? What do you think?

**Review by patryk-sierzega — APPROVED** [review]

**Pull request merged**